packages/lib-sourcify/test/Compilation/SolidityCompilation.spec.ts

```diff
@@ -0,0 +1,266 @@
+import { describe, it } from 'mocha';
+import { expect, use } from 'chai';
+import path from 'path';
+import fs from 'fs';
+import { SolidityCompilation } from '../../src/Compilation/SolidityCompilation';
+import { solc } from '../utils';
+import {
+  CompilationTarget,
+  Metadata,
+} from '../../src/Compilation/CompilationTypes';
+import { SolidityJsonInput } from '../../src/Compilation/SolidityTypes';
+import chaiAsPromised from 'chai-as-promised';
+
+use(chaiAsPromised);
+
+function getSolcSettingsFromMetadata(metadata: Metadata) {
+  const metadataSettings = JSON.parse(JSON.stringify(metadata.settings));
+  delete metadataSettings.compilationTarget;
+  return metadataSettings;
+}
+
+function getCompilationTargetFromMetadata(
+  metadata: Metadata,
+): CompilationTarget {
+  const path = Object.keys(metadata.settings.compilationTarget)[0];
+  const name = metadata.settings.compilationTarget[path];
+  return {
+    name,
+    path,
+  };
+}
+
+describe.only('SolidityCompilation', () => {
+  it('should compile a simple contract', async () => {
+    const contractPath = path.join(__dirname, '..', 'sources', 'Storage');
+    const metadata = JSON.parse(
+      fs.readFileSync(path.join(contractPath, 'metadata.json'), 'utf8'),
+    );
+    const sources = {
+      'project:/contracts/Storage.sol': {
+        content: fs.readFileSync(
+          path.join(contractPath, 'sources', 'Storage.sol'),
+          'utf8',
+        ),
+      },
+    };
+
+    const compilation = new SolidityCompilation(
+      solc,
+      metadata.compiler.version,
+      {
+        language: 'Solidity',
+        sources,
+        settings: getSolcSettingsFromMetadata(metadata),
```

Comment:
This part of the test code is quite redundant and can be refactored IMO

Reply:
I still have to refactor the tests, right now I just wanted to no-brainer test as much lines / branches as possible